scripts/generate_parser_test_files.py

```diff
@@ -5,25 +5,25 @@
 
 from tqdm import tqdm
 
-from fundus import BaseCrawler, Crawler, PublisherCollection
+from fundus import Crawler, PublisherCollection
 from fundus.logging import basic_logger
 from fundus.publishers.base_objects import PublisherEnum
 from fundus.scraping.article import Article
-from fundus.scraping.html import FundusSource
-from fundus.scraping.scraper import Scraper
+from fundus.scraping.filter import RequiresAll
+from fundus.scraping.html import WebSource
+from fundus.scraping.scraper import BaseScraper, WebScraper
 from tests.test_parser import attributes_required_to_cover
 from tests.utility import HTMLTestFile, get_test_case_json, load_html_test_file_mapping
 
 
 def get_test_article(enum: PublisherEnum, url: Optional[str] = None) -> Optional[Article]:
-    crawler: BaseCrawler
     if url is None:
         crawler = Crawler(enum)
+        return next(crawler.crawl(max_articles=1, error_handling="suppress", only_complete=True), None)
     else:
-        source = FundusSource([url], publisher=enum.publisher_name)
-        scraper = Scraper(source, parser=enum.parser)
-        crawler = BaseCrawler(scraper)
-    return next(crawler.crawl(max_articles=1, error_handling="suppress", only_complete=True), None)
+        source = WebSource([url], publisher=enum.publisher_name)
+        scraper = BaseScraper(source, parser_mapping={enum.publisher_name: enum.parser})
+        return next(scraper.scrape(error_handling="suppress", extraction_filter=RequiresAll()), None)
```

Comment:
The `else` is not necessary here, and can be removed to save indentation.

Reply:
That is in fact true 😄 hmm, somehow I think the else makes the code more readable. What do you think?

Reply:
I guess it's preference regarding what is more readable. IMO less indentation usually results in more readable code. Ruff also does this transformation automatically, if we decide to use it someday: https://docs.astral.sh/ruff/rules/superfluous-else-return/

But in the end, this is not critical, leave it in if you like the else more.

Reply:
I agree. You had me with `ruff`.